pkg/process/util/network_filter.go

```diff
@@ -0,0 +1,102 @@
+package util
+
+import (
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// ConnectionFilter holds a user-defined blacklisted IP/CIDR, and ports
+type ConnectionFilter struct {
+	IP       *net.IPNet
+	Ports    map[uint16]struct{}
+	AllPorts bool
+}
+
+// ParseConnectionFilters takes the user defined blacklist and returns a slice of ConnectionFilters
+func ParseConnectionFilters(filters map[string][]string) (blacklist []*ConnectionFilter) {
+	for ip, ports := range filters {
+		filter := &ConnectionFilter{Ports: map[uint16]struct{}{}}
+		var subnet *net.IPNet
+		var err error
+
+		// retrieve valid IPs
+		if strings.ContainsRune(ip, '*') {
+			subnet = nil // use for wildcard
+		} else if strings.ContainsRune(ip, '/') {
+			_, subnet, err = net.ParseCIDR(ip)
+		} else if strings.ContainsRune(ip, '.') {
+			_, subnet, err = net.ParseCIDR(ip + "/32") // if given ipv4, prefix length of 32
+		} else if strings.Contains(ip, "::") {
+			_, subnet, err = net.ParseCIDR(ip + "/64") // if given ipv6, prefix length of 64
+		} else {
+			log.Debugf("Invalid IP/CIDR/* defined for connection filter: %s", err)
+			continue
+		}
+
+		if err != nil {
+			log.Debugf("Could not parse given IPs: %s", err)
```

Comment:
Let's error log this, and explain that the rule will not be respected since a user would want to know about it :)